marlowe/doc/playground-overview.adoc

```diff
@@ -19,119 +19,172 @@ contract in the browser-based tool, the Marlowe Playground, a web tool
-Marlowe contracts transaction by transaction. Typically these are
-contracts developed in the Haskell editor or within Blockly, but they can also be entered and modified in the editor pane here too.
+On successful compilation the result can be sent to the simulator or to Blockly: these options are provided by the *Send to Simulator* and *Send to Blockly* buttons in the top RH corner of the page.
 
-image:simulation-tab.png[The Simulation tab]
+If the contract cannot successfully be converted to Marlowe, the errors are also shown in overlay:
 
-The simulation tab is selected on the left-hand side of the screen (number 1 in the figure). At the foot of this side of the screen, and on all tabs, is a link to the online tutorial for Marlowe (number 2). The pane also contains a number of example contracts, and these are selected above the main editor window (number 3).
+image:haskell-errors.png[Errors in compiling Haskell code to Marlowe]
 
-Contracts in the simulator can be transferred to the Blockly visual editor (button number 4), and if you have a GitHub account, it's also possible to save contracts that you develop as _GitHub gists_ (number 5). To do this, log into your account using the  *Log in* button at the top right of the screen. You will then see buttons to allow you to *Publish* or save contracts as gists, and to *Load* contracts from GitHub.
 
-The tile at the foot of the window (number 6) gives information about the current contract and its state. The tab shown here will give information about the state of execution of the Marlowe contract, including the values held in the various accounts of the contract, choices and payments made, the current block number and so on. Other tabs give access to static analysis and its results, as well as other warnings and errors.
+=== The JavaScript Editor: developing embedded contracts
 
-In simulating a contract there are two stages
+The editor supports the development of Marlowe contracts described in JavaScript, too. We can use JavaScript to make contract definitions more readable by using JS definitions for sub-components, abbreviations, and simple template functions. The JS editor is shown in the following image.
 
-* Inputs are chosen, and form part of
-* Transactions that aggregate a number of inputs and payments.
+image:js-editor.png[The JavaScript editor]
 
-Inputs are chosen (number 7) from the inputs that are relevant at the particular point of the contract. By relevant here we mean that these are the (only) inputs that will make the contract progress at this stage. In other words, input choice is "intelligent". An input is added to the transaction that is being built by selecting the *+* button (number 7).
+The JS editor is open here on the Escrow example contained in the examples. To describe a Marlowe contract in the editor, the _last expression in the file_ needs to be of type  `Contract`: here we define the top-level constant `contract` to be of that type. 
```

Comment:
This is not true at the moment, you have to set the variable `contract`, setting the last expression doesn't work. I have already updated the JS section of the tutorial

Reply:
Rewritten the description here, Thanks for the correction!